File: app/src/debug/java/com/handstandsam/maintainableespresso/DebugBroadcastReceiver.java
package com.handstandsam.maintainableespresso;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;

import com.handstandsam.maintainableespresso.utils.IntentUtils;

import timber.log.Timber;

public class DebugBroadcastReceiver extends BroadcastReceiver {

    private static final String PR_NOTIFICATION = "PR_NOTIFICATION";


    public DebugBroadcastReceiver() {
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        handleIntent(context, intent);
    }

    void handleIntent(Context context, @NonNull Intent intent) {
        String intentAction = intent.getAction();

        if (intentAction == null || !PR_NOTIFICATION.equals(intentAction)) {
            return;
        }
        Timber.d("processing " + PR_NOTIFICATION);

        Bundle extras = intent.getExtras();
        if (extras == null) {
            return;
        }

        final String username = extras.getString("username");
        Timber.d("username " + username);

        IntentUtils.triggerPRNotification(context, username);
    }
}

File: app/src/debug/java/com/handstandsam/shoppingapp/DebugBroadcastReceiver.kt
package com.handstandsam.shoppingapp

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.handstandsam.shoppingapp.utils.IntentUtils
import timber.log.Timber


class DebugBroadcastReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        handleIntent(context, intent)
    }

    private fun handleIntent(context: Context, intent: Intent) {
        val intentAction = intent.action

        if (intentAction == null || PR_NOTIFICATION != intentAction) {
            return
        }
        Timber.d("processing " + PR_NOTIFICATION)

        val extras = intent.extras ?: return

        val username = extras.getString("username")
        Timber.d("username " + username!!)

        IntentUtils.triggerPRNotification(context, username)
    }

    companion object {

        private val PR_NOTIFICATION = "PR_NOTIFICATION"
    }
}
